unix/c_editline.c

```diff
@@ -33,6 +33,50 @@ void go_libedit_set_clientdata(EditLine *el, int v) {
     el_set(el, EL_CLIENTDATA, p);
 }
 
+/************* terminals ************/
+
+const char term_me_ansi[] = "\033[0m";
+const char term_bp_off[] = "\033[?2004l";
+const char term_bp_on[] = "\033[?2004h";
+const char term_bp_start[] = "\033[200~";
+const char term_bp_end[] = "\033[201~";
+
+int go_libedit_term_supports_bracketed_paste(EditLine* el) {
+    // termcap has no entry for bracketed paste, so we use the me cap as a
+    // proxy. If the me cap uses the ANSI control code, the terminal will either
```

Comment:
"If the me cap uses the ANSI control code, the terminal will either support or ignore" -> what is the reasoning here?

1) of the 85 entries in my /etc/termcap that do have a `me` cap, only 7 use `\E[0m`. The rest uses `\E[m`.

2) I recall that the `\E[` prefix is markedly older than the ansi spec. I wouldn't trust its presence to be a guarantee that `\E[?2004h` and conversely `\E[200~` will not mess up something. 

I am ok with a test based on "likelihood of support" but I notice that https://cirw.in/blog/bracketed-paste says that bracketed mode was implemented in more "modern" terminals. I'm not sure there is a reliable heuristic using the available termcaps.

What about a test on TERM? it seems to me that these terminals dear to you report their compatibility with TERM=xterm. You could add a couple known citizens alongside that (things starting with "xterm", ending with "rxvt", containing tmux/screen, `st`, etc).

Reply:
> If the me cap uses the ANSI control code, the terminal will either support or ignore" -> what is the reasoning here?

The reasoning is that ANSI control codes have a structure ("CSI", parameter bytes, intermediate bytes, then the final byte) that makes it possible for a terminal to ignore control codes they don't recognize. So I figure if a terminal is using an ANSI-like control code we're probably safe to plow ahead.

> of the 85 entries in my /etc/termcap that do have a me cap, only 7 use \E[0m. The rest uses \E[m.

Ugh. Those two control codes are equivalent because omitted parameters default to 0. I've updated the check to verify only that the code starts with `\E[`.

> I recall that the \E[ prefix is markedly older than the ansi spec. I wouldn't trust its presence to be a guarantee that \E[?2004h and conversely \E[200~ will not mess up something.

Yeah, probably, but how much do we care about a terminal that implements `\E[` in an ASCII-noncompliant way?

> What about a test on TERM? it seems to me that these terminals dear to you report their compatibility with TERM=xterm. You could add a couple known citizens alongside that (things starting with "xterm", ending with "rxvt", containing tmux/screen, st, etc).

Wanted to avoid having to come up with the whitelist, but it sounds like you already know the whitelist. Screen at least doesn't support bracketed paste, so we can avoid that one. Not sure about tmux. Perhaps just rxvt and xterm would be sufficient.

Reply:
> Screen at least doesn't support bracketed paste

That kinda validates my point that support for `\E[` is not sufficient.
Doest it at least properly absorb / kill the bracked paste enablement sequence?

Reply:
> Doest it at least properly absorb / kill the bracked paste enablement sequence?

It does.

In any case, updated to use a `TERM`-based detection approach.